src/challenge-status/challenge-status.resolver.ts

```diff
@@ -0,0 +1,36 @@
+import { Resolver, Args, Query } from '@nestjs/graphql';
+import { UseGuards } from '@nestjs/common';
+import { ChallengeStatusEntity as ChallengeStatus } from './entities/challenge-status.entity';
+import { GqlJwtAuthGuard } from '../common/guards/gql-jwt-auth.guard';
+import { ChallengeStatusService } from './challenge-status.service';
+import { State } from './entities/state.enum';
+import UpdateChallengeStatus from './dto/update-challenge-status.dto';
+
+@Resolver()
+export class ChallengeStatusResolver {
+  constructor(private challengeStatusService: ChallengeStatusService) {}
+
+  @Query(() => ChallengeStatus)
```

Comment:
Can you explain why this is a Query? and why we need it?

Reply:
Here I followed your comment from 9d ago "ChallengeStatus should not allow direct mutations. It is managed internally, so just Queries." My idea/view on that was that when a player interacts with the challenge, then the interactions are hitting an endpoint/sending a query which trigger the updates and creations in the repo. Should they just be methods with guards then?

Reply:
I think I was not clear (again). When I said "ChallengeStatus should not allow direct mutations.", I  mean the API should not expose methods that allow to set a challenge as completed (for instance). Why? Because, that is a secondary effect of an Accepted submission, and is updated internally (using `ChallengeStatusService`) when the result of a submission is received.

Nevertheless, we want to be able to ask the API about the status, right? That's the meaning of "so just Queries".